10 Notes/Component Specification - Prover.md

```diff
@@ -0,0 +1,388 @@
+| ------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------- |
+| `SampleAndProve`         | Main component in the circuit. Verifies `nSamples` cells (`cell->block->slot`) and the `slot->dataset` path, binding the proof to `dataSetRoot`, `slotIndex`, and `entropy`. | `maxDepth, maxLog2NSlots, blockTreeDepth, nFieldElemsPerCell, nSamples` | `entropy`, `dataSetRoot`, `slotIndex`, `slotRoot`, `nCellsPerSlot`, `nSlotsPerDataSet`, `slotProof[maxLog2NSlots]`, `cellData[nSamples][nFieldElemsPerCell]`, `merklePaths[nSamples][maxDepth]` | -                     |
+| `ProveSingleCell`        | Verifies one sampled cell: hashes `cellData` with Poseidon2 and checks the concatenated Merkle path up to `slotRoot`.                                                        | `nFieldElemsPerCell, botDepth, maxDepth`                                | `slotRoot`, `data[nFieldElemsPerCell]`, `lastBits[maxDepth]`, `indexBits[maxDepth]`, `maskBits[maxDepth+1]`, `merklePath[maxDepth]`                                                             | -                     |
+| `RootFromMerklePath`     | Reconstructs a Merkle root from a leaf and path using `KeyedCompression`.                                                                                                    | `maxDepth`                                                              | `leaf`, `pathBits[maxDepth]`, `lastBits[maxDepth]`, `maskBits[maxDepth+1]`, `merklePath[maxDepth]`                                                                                              | `recRoot`             |
+| `CalculateCellIndexBits` | Derives the index bits for a sampled cell from `(entropy, slotRoot, counter)`, masked by `cellIndexBitMask`.                                                                 | `maxLog2N`                                                              | `entropy`, `slotRoot`, `counter`, `cellIndexBitMask[maxLog2N]`                                                                                                                                  | `indexBits[maxLog2N]` |
+
+
+**Circuit Utility Templates**
```

Comment:
It seems like almost everything in the table below had input "input" and output "output". That's not very insightful. Maybe it would be more useful to describe in some cases what the input and output is, rather than what the variables are called.